File: miscellaneous/graph-bfs/20230108/graph_bfs.py
from typing import List, Set


def graph_bfs(graph: List[List[int]]) -> List[int]:
    values: List[int] = []
    queue = [0]
    seen = {0}

    while (len(queue)):
        vertex = queue.pop(0)
        values.append(vertex)

        for connection in graph[vertex]:
            if connection not in seen:
                seen.add(connection)
                queue.append(connection)

    return values

File: miscellaneous/graph-bfs/20230108/graph-bfs.php
<?php

declare(strict_types=1);

function graphBfs(array &$graph): array
{
    $values = [];
    $queue = [0];

    while (count($queue)) {
        $vertex = array_shift($queue);
        $values[] = $vertex;

        array_map(function (int $connection) use (&$values, &$queue) {
            !in_array($connection, $values) && $queue[] = $connection;
        }, $graph[$vertex]);
    }

    return $values;
}